mitodl/micromasters
# Removed privacy page from signup flow

#### What are the relevant tickets?

Closes #899 

#### What's this PR do?

Removes the privacy page (4th in order) from the signup flow

#### How should this be manually tested?

Set filled_out to `false` for your user and run through the signup flow. Make sure validation still works. The privacy form in the settings page should also be tested.

#### Any background context you want to provide?

I also fixed up some js tests that used `Object.assign({}, USER_PROFILE_RESPONSE, {...})`. Object.assign can produce some [unexpected behavior with deep cloning](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) (see "Warning for Deep Clone") and I had a very nasty time with a js test last week as a result of it
